state-synchronizer/src/executor_proxy.rs

```diff
@@ -151,4 +166,58 @@ impl ExecutorProxyTrait for ExecutorProxy {
         );
         Ok(waypoint_li.clone())
     }
+
+    async fn publish_on_chain_config_updates(&mut self, events: Vec<ContractEvent>) -> Result<()> {
+        if events.is_empty() {
+            return Ok(());
+        }
+
+        let access_paths = ON_CHAIN_CONFIG_REGISTRY
+            .iter()
+            .map(|config_id| config_id.access_path())
+            .collect();
+        let curr_configs = self
+            .storage_read_client
+            .batch_fetch_config(access_paths)
+            .await;
+
+        let mut new_configs_map = HashMap::new();
+        for (id, configs) in ON_CHAIN_CONFIG_REGISTRY.iter().zip(curr_configs.iter()) {
+            new_configs_map.insert(id, configs);
+        }
+
+        // calculate deltas
+        let mut changed_configs = HashSet::new();
+        let mut new_configs = HashMap::new();
```

Comment:
why we have a `new_configs_map` and `new_configs`?

Reply:
rip unused code.. I'm surprised linter did not catch this